src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -22,9 +25,28 @@ public class GenerateEqualsAndGetHashCodeFromMembersTests : AbstractCSharpCodeAc
         private static readonly TestParameters CSharp6 =
             new TestParameters(parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
 
+        private TestParameters CSharp6Implicit => CSharp6.WithOptions(ImplicitTypingEverywhere());
+        private TestParameters CSharp6Explicit => CSharp6.WithOptions(ExplicitTypingEverywhere());
+
         protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
             => new GenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider((IPickMembersService)parameters.fixProviderData);
 
+        private readonly CodeStyleOption<bool> onWithInfo = new CodeStyleOption<bool>(true, NotificationOption.Suggestion);
+        private readonly CodeStyleOption<bool> offWithInfo = new CodeStyleOption<bool>(false, NotificationOption.Suggestion);
+
+        private IDictionary<OptionKey, object> ExplicitTypingEverywhere() =>
+            OptionsSet(
+                SingleOption(CSharpCodeStyleOptions.VarElsewhere, offWithInfo),
+                SingleOption(CSharpCodeStyleOptions.VarWhenTypeIsApparent, offWithInfo),
+                SingleOption(CSharpCodeStyleOptions.VarForBuiltInTypes, offWithInfo));
+
+        private IDictionary<OptionKey, object> ImplicitTypingEverywhere() =>
+            OptionsSet(
+                SingleOption(CSharpCodeStyleOptions.VarElsewhere, onWithInfo),
+                SingleOption(CSharpCodeStyleOptions.VarWhenTypeIsApparent, onWithInfo),
+                SingleOption(CSharpCodeStyleOptions.VarForBuiltInTypes, onWithInfo));
```

Comment:
❔ Can we just use the fields from AbstractUseTypeRefactoringTests instead of defining new ones here?

Reply:
Not exactly.  there's instance behavior here needed.  I was able to share more code by directly deriving from AbstractUseTypeRefactoringTests  though.  It's a bit strange, but not too bad.

Reply:
> It's a bit strange, but not too bad.

I agree with this conclusion.

Reply:
because this was needed in several other locatoins (including both refactoring and diagnostic tests) i actually extracted these out into extension methods that any C# test can call.  This was necessary as there is no common *c#* base class for all tehse tests.  the only common class is in a language agnostic level, and then the two hierarchies split to the diagnostic/refactoring subclasses before we get to the language level.

It is pretty clean, and it means any refactoring/diagnostic test in the C# level can easily ask for these options.